### ISSUE TITLE
Skip environments that don't match grep expression

### ISSUE THREAD

**Issue by schontz** [theintern/intern#950]
I have node and browser tests. In my node tests, I have a suite named `Config`. When I only want to run those tests I execute the following:

```
npx intern grep=Config
```
## Expected behavior

Only node tests should run and my browser tunnels should not load up.

## Current behavior

My browser tunnel loads up even though all the tests are skipped. There is a significant performance hit. @jason0x43  explains:
> Currently, grep is evaluated as suites are run rather than before testing starts, so Intern won't know that all tests in a suite have been filtered out before it needs to start a remote.

## Possible solution

For now, you have to manually set the environments variable:

```
npx intern environments=node grep=Config
```

**Comment by jason0x43** [comment]
Probably the best way to attack this would be to evaluate grep as tests are added (i.e., don't add tests that don't match the grep expression) rather than as they're run.

**Comment by schontz** [comment]
Related to this, if no tests in a suite match the grep condition, none of the before/after functions should run either.

**Comment by devpaul** [comment]
Hi @schontz,

I've taken a look at this and unfortunately it's not really possible to use `grep` to determine what tests should be run before the browser tunnels are loaded. This is because `grep` works on test's ID's, which are loaded by the browser's loader. There's no universally reliable way to load these test names in node.

We did discuss one possible solution. What if you were able to filter suites by providing a part of their filename and if no suites were to run for the browser then the tunnel wouldn't start? It would look something like `intern only=Config` and only suites that had "Config" in their filename would run. By operating on filenames rather than test ids we could identify whether any suites would be sent to the browser before starting a tunnel. Would this address your use case?

Let me know what you think.

**Comment by schontz** [comment]
I am not on said project anymore, but having an `only` option would address the use case I was looking at. I don't know if it would be useful for others.

**Comment by jason0x43** [comment]
Closing in favor of #1022.
